api/v1alpha1/monitoringconfig_types.go

```diff
@@ -0,0 +1,79 @@
+// everest-operator
+// Copyright (C) 2022 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MonitoringType is a type of monitoring.
+type MonitoringType string
+
+const (
+	// PMM represents monitoring via PMM.
+	PMM MonitoringType = "pmm"
+)
+
+// MonitoringConfigSpec defines the desired state of MonitoringConfig.
+type MonitoringConfigSpec struct {
+	// Type is type of monitoring.
+	// +kubebuilder:validation:Enum=pmm
+	Type MonitoringType `json:"type"`
+	// CredentialsSecretName is the name of the secret with credentials.
+	CredentialsSecretName string `json:"credentialsSecretName"`
+	// Resources defines resource limitations for the monitoring.
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
I don't currently have an answer for this but I wanted to start the discussion.
Should the `Resources` be part of the `MonitoringConfig` or should it be part of the `DatabaseCluster` that references it?
Phrasing this in a different way, would we ever need to set different resource limits for the same Monitoring instance? Would a PMM client container need more resources in DatabaseCluster A than in DatabaseCluster B?

Reply:
I think you're right. This shall sit with the `DatabaseCluster` and not `MonitoringConfig`.
Different DBs may require more/less resources.